src/common/marlin_vars.hpp

```diff
@@ -99,48 +102,48 @@ enum {
     MARLIN_VAR_INDEX_E = 3,
 };
 
-static const uint8_t MARLIN_VAR_MOTION_MSK_X = 1 << MARLIN_VAR_INDEX_X;
-static const uint8_t MARLIN_VAR_MOTION_MSK_Y = 1 << MARLIN_VAR_INDEX_Y;
-static const uint8_t MARLIN_VAR_MOTION_MSK_Z = 1 << MARLIN_VAR_INDEX_Z;
-static const uint8_t MARLIN_VAR_MOTION_MSK_E = 1 << MARLIN_VAR_INDEX_E;
-
-static const uint32_t TIME_TO_END_INVALID = (uint32_t)-1;
-
-typedef enum {
-    mpsIdle = 0,
-    mpsWaitGui,
-    mpsPrintPreviewInit,
-    mpsPrintPreviewLoop,
-    mpsPrintInit,
-    mpsPrinting,
-    mpsPausing_Begin,
-    mpsPausing_Failed_Code,
-    mpsPausing_WaitIdle,
-    mpsPausing_ParkHead,
-    mpsPaused,
-    mpsResuming_Begin,
-    mpsResuming_Reheating,
-    mpsResuming_UnparkHead_XY,
-    mpsResuming_UnparkHead_ZE,
-    mpsAborting_Begin,
-    mpsAborting_WaitIdle,
-    mpsAborting_ParkHead,
-    mpsAborted,
-    mpsFinishing_WaitIdle,
-    mpsFinishing_ParkHead,
-    mpsFinished,
-    mpsExit, // sets idle, notifies clients to close fsm
-    mpsCrashRecovery_Begin,
-    mpsCrashRecovery_Retracting,
-    mpsCrashRecovery_Lifting,
-    mpsCrashRecovery_XY_Measure,
-    mpsCrashRecovery_XY_HOME,
-    mpsCrashRecovery_Axis_NOK,
-    mpsCrashRecovery_Repeated_Crash,
-    mpsPowerPanic_acFault,
-    mpsPowerPanic_Resume,
-    mpsPowerPanic_AwaitingResume,
-} marlin_print_state_t;
+static constexpr uint8_t MARLIN_VAR_MOTION_MSK_X = 1 << MARLIN_VAR_INDEX_X;
+static constexpr uint8_t MARLIN_VAR_MOTION_MSK_Y = 1 << MARLIN_VAR_INDEX_Y;
+static constexpr uint8_t MARLIN_VAR_MOTION_MSK_Z = 1 << MARLIN_VAR_INDEX_Z;
+static constexpr uint8_t MARLIN_VAR_MOTION_MSK_E = 1 << MARLIN_VAR_INDEX_E;
+
+static constexpr uint32_t TIME_TO_END_INVALID = (uint32_t)-1;
+
+enum class marlin_print_state_t : uint8_t {
```

Comment:
When we are touching the type and moving everything to C++, I might as well poke into the naming conventions.

We kind of agreed types should be `CamelCase`. I know specifically the marlin and around it, this is historically different, so I'm not pushing too strongly for that, but when changing it to enum class and effectively enforcing everything to prefix it with the type, it sounds like a good time to fix the name.

Similarly, the prefix and suffix makes it really long in many places. The `_t` is a `C` relic and isn't usually used in C++. And, C++ can prefer to use a namespace instead of prefix ‒ for example `marlin::PrintState`.

Reply:
it should be print processor independent (GUI should not know about marlin ... mostly)
So I will use `PrintState`